Sources/NIO/EventLoop.swift

```diff
@@ -729,6 +742,9 @@ final public class MultiThreadedEventLoopGroup: EventLoopGroup {
 
         g.notify(queue: q) {
             let failure = self.eventLoops.map { try? $0.close0() }.filter { $0 == nil }.count > 0
+
+            // TODO: In the next major release we should join in the Thread used by the EventLoop before invoking the callback to ensure
+            //       it is really gone.
```

Comment:
No need to change  this PR, but `ThreadSpecificVariable` is also going to need to change to properly call pthread_key_delete inside the destructor function passed to pthread_key_create. I ran into this working on #133 but couldn't wrap my head around how to fix it.

Reply:
@weissi can you take a look ?

Reply:
@toffaletti / @normanmaurer using `pthread_key_delete` is unfortunately not straightforward at all I think. I'll create an issue.